xonsh/aliases.py

```diff
@@ -665,44 +678,32 @@ def xexec(args, stdin=None):
     explicitly with ![exec command]. For more details, please see
     http://xon.sh/faq.html#exec.
     """
-    if len(args) == 0:
-        return (None, "xonsh: exec: no args specified\n", 1)
-
-    parser = argparse.ArgumentParser(add_help=False)
-    parser.add_argument("-h", "--help", action="store_true")
-    parser.add_argument("-l", dest="login", action="store_true")
-    parser.add_argument("-c", dest="clean", action="store_true")
-    parser.add_argument("-a", dest="name", nargs="?")
-    parser.add_argument("command", nargs=argparse.REMAINDER)
-    args = parser.parse_args(args)
-
-    if args.help:
-        return inspect.getdoc(xexec)
-
-    if len(args.command) == 0:
+    if len(command) == 0:
         return (None, "xonsh: exec: no command specified\n", 1)
 
-    command = args.command[0]
-    if args.name is not None:
-        args.command[0] = args.name
-    if args.login:
-        args.command[0] = "-{}".format(args.command[0])
+    cmd = command[0]
+    if name:
+        command[0] = name
+    if login:
+        command[0] = "-{}".format(command[0])
 
     denv = {}
-    if not args.clean:
+    if not clean:
         denv = XSH.env.detype()
 
     try:
-        os.execvpe(command, args.command, denv)
+        os.execvpe(cmd, command, denv)
     except FileNotFoundError as e:
         return (
             None,
-            "xonsh: exec: file not found: {}: {}"
-            "\n".format(e.args[1], args.command[0]),
+            "xonsh: exec: file not found: {}: {}" "\n".format(e.args[1], command[0]),
             1,
         )
 
 
+xexec = ArgParserAlias(func=xexec_fn, has_args=True, prog="xexec")
```

Comment:
Why not call the function `xexec` and overwrite the variable so that `prog` is deduced?

Reply:
the xexec_fn can be used during tests and other pure Python scripts. Also explicitly we are defining the object.